e2e/chat.spec.ts

```diff
@@ -493,7 +493,7 @@ describe('chat', () => {
 
     await chat.goto();
 
-    const searchButton = page.getByTestId('search-conversation-button');
+    const searchButton = page.getByRole('button', { name: 'Search History' });
```

Comment:
I like this type of change. One thing playwright and react testing library do well is they give you selectors you should be able to use without having to use data-testid. Using stuff like getByRole ensures that we've built semantically correct, screen reader friendly code. I'd actually argue in most cases, if you can't use something other than getByTestId then we should actually change how the code is built.